tests/test_routing.py

```diff
@@ -516,6 +524,8 @@ def test_host_reverse_urls() -> None:
         )
         == "https://port.example.org:3600/"
     )
+    with pytest.raises(NoMatchFound):
+        mixed_hosts_app.url_path_for("api", path="whatever", foo="bar")
```

Comment:
Testing the circumstance where there are some remaining parameters and an exception is expected.
https://github.com/encode/starlette/blob/master/starlette/routing.py#L532

Reply:
I passed an extra parameter `foo` in this case. This will make the if statement become `False` and jump to this line:
```
raise NoMatchFound(name, path_params)
```
This behavior was not tested and covered in the other tests before.